src/reference/asciidoc/rsocket.adoc

```diff
@@ -0,0 +1,327 @@
+[[rsocket]]
+== RSocket Support
+
+The RSocket Spring Integration module (`spring-integration-rsocket`) allows for executions of https://rsocket.io/[RSocket application protocol].
+
+You need to include this dependency into your project:
+
+====
+.Maven
+[source, xml, subs="normal"]
+----
+<dependency>
+    <groupId>org.springframework.integration</groupId>
+    <artifactId>spring-integration-rsocket</artifactId>
+    <version>{project-version}</version>
+</dependency>
+----
+
+.Gradle
+[source, groovy, subs="normal"]
+----
+compile "org.springframework.integration:spring-integration-rsocket:{project-version}"
+----
+====
+
+This module is available starting version 5.2 and is fully based on the Spring Messaging foundation with its RSocket components implementation, such as `RSocketRequester`, `RSocketMessageHandler` and `RSocketStrategies`.
+
+Before starting an integration flow processing via channel adapters we need to establish RSocket connection between server and client.
+For this purpose Spring Integration RSocket support provides a `ServerRSocketConnector` and `ClientRSocketConnector` implementations of the `AbstractRSocketConnector`.
+
+The `ServerRSocketConnector` exposes a listener on the host and port according to provided `io.rsocket.transport.ServerTransport` for accepting connections from clients.
+An internal `RSocketFactory.ServerRSocketFactory` instance can be customized with the `setFactoryConfigurer()`, as well as other options can be configured, e.g. `RSocketStrategies` and `MimeType` for payload data and headers metadata.
+A connected client is stored as a `RSocketRequester` under the key determined by the `clientRSocketKeyStrategy` `BiFunction<Map<String, Object>, DataBuffer, Object>`.
+By default a connect data is used for key as converted to string with UTF-8 charset.
+Such an `RSocketRequester` registry can be used in the application logic to determine a particular client connection for interaction exactly with this one or for publishing the same message to all connected clients.
+When connection is established from the client, an `RSocketConnectedEvent` is emitted from the `ServerRSocketConnector`.
+This is similar to what is provided by the `@ConnectMapping` annotation in Spring Messaging module.
+The mapping pattern is `*` - means accept all the client routes.
+The `RSocketConnectedEvent` could be used to distinguish different routes via `DestinationPatternsMessageCondition.LOOKUP_DESTINATION_HEADER` header.
+
+The typical server configuration may look like this:
+
+====
+[source, java]
+----
+@Bean
+public RSocketStrategies rsocketStrategies() {
+    return RSocketStrategies.builder()
+        .decoder(StringDecoder.allMimeTypes())
+        .encoder(CharSequenceEncoder.allMimeTypes())
+        .dataBufferFactory(new NettyDataBufferFactory(PooledByteBufAllocator.DEFAULT))
+        .build();
+}
```

Comment:
Note that these are `RSocketStrategies` defaults now to register basic codecs and use pooled buffers. So the above is equivalent to `RSocketStrategies.create()` or `RSocketStrategies.builder.build()`.

Reply:
Yeah... I've noticed that during refactoring in the main code, but forgot to change this snippet to something what is not default.
Thank you for the pointer, @rstoyanchev !